plugins/Compressor/Compressor.cpp

```diff
@@ -0,0 +1,652 @@
+					{
+						const float temp1 = qMin(gainResult[0], gainResult[1]);
+						gainResult[0] = linearInterpolate(gainResult[0], temp1, blend);
+						gainResult[1] = linearInterpolate(gainResult[1], temp1, blend);
+					}
+					else if (blend <= 2)// Blend to average volume
+					{
+						const float temp1 = qMin(gainResult[0], gainResult[1]);
+						const float temp2 = (gainResult[0] + gainResult[1]) * 0.5f;
+						gainResult[0] = linearInterpolate(temp1, temp2, blend - 1);
+						gainResult[1] = gainResult[0];
+					}
+					else// Blend to maximum volume
+					{
+						const float temp1 = (gainResult[0] + gainResult[1]) * 0.5f;
+						const float temp2 = qMax(gainResult[0], gainResult[1]);
+						gainResult[0] = linearInterpolate(temp1, temp2, blend - 2);
+						gainResult[1] = gainResult[0];
+					}
+				}
+				break;
+			}
+		}
+
+		// Bias compression to the left or right (or mid or side)
+		if (stereoBalance != 0)
+		{
+			gainResult[0] = 1 - ((1 - gainResult[0]) * (stereoBalance > 0 ? 1 - stereoBalance : 1));
+			gainResult[1] = 1 - ((1 - gainResult[1]) * (stereoBalance < 0 ? 1 + stereoBalance : 1));
+		}
+
+		// For visualizer
+		m_displayGain[0] = gainResult[0];
+		m_displayGain[1] = gainResult[1];
+
+		// Delay the signal by 20 ms via ring buffer if lookahead is enabled
+		if (lookahead)
+		{
+			++m_inputBufLoc;
+			if (m_inputBufLoc >= m_lookaheadDelayLength)
+			{
+				m_inputBufLoc = 0;
+			}
+
+			const float temp[2] = {drySignal[0], drySignal[1]};
+			s[0] = m_inputBuf[0][m_inputBufLoc];
+			s[1] = m_inputBuf[1][m_inputBufLoc];
+
+			m_inputBuf[0][m_inputBufLoc] = temp[0];
+			m_inputBuf[1][m_inputBufLoc] = temp[1];
+		}
+		else
+		{
+			s[0] = drySignal[0];
+			s[1] = drySignal[1];
+		}
+
+		float delayedDrySignal[2] = {s[0], s[1]};
+		
+		if (midside)// Convert left/right to mid/side
+		{
+			const float temp = s[0];
+			s[0] = (temp + s[1]) * 0.5;
+			s[1] = temp - s[1];
+		}
+
+		s[0] *= gainResult[0] * m_inGainVal * m_outGainVal * (outBalance > 0 ? 1 - outBalance : 1);
+		s[1] *= gainResult[1] * m_inGainVal * m_outGainVal * (outBalance < 0 ? 1 + outBalance : 1);
+
+		if (midside)// Convert mid/side back to left/right
+		{
+			const float temp1 = s[0];
+			const float temp2 = s[1] * 0.5;
+			s[0] = temp1 + temp2;
+			s[1] = temp1 - temp2;
+		}
+
+		m_prevOut[0] = s[0];
+		m_prevOut[1] = s[1];
+
+		// Negate wet signal from dry signal
+		if (audition)
+		{
+			s[0] = (-s[0] + delayedDrySignal[0] * m_outGainVal);
+			s[1] = (-s[1] + delayedDrySignal[1] * m_outGainVal);
+		}
+		else if (autoMakeup)
+		{
+			s[0] *= m_autoMakeupVal;
+			s[1] *= m_autoMakeupVal;
+		}
+
+		// Calculate wet/dry value results
+		const float temp1 = delayedDrySignal[0];
+		const float temp2 = delayedDrySignal[1];
+		buf[f][0] = d * temp1 + w * s[0];
+		buf[f][1] = d * temp2 + w * s[1];
+		buf[f][0] = (1 - m_mixVal) * temp1 + m_mixVal * buf[f][0];
+		buf[f][1] = (1 - m_mixVal) * temp2 + m_mixVal * buf[f][1];
+
+		outSum += s[0] + s[1];
```

Comment:
`outSum` should be computed with the squares of the wet/dry mixed values. See the amplifier code: https://github.com/LMMS/lmms/blob/00ac4f58f50d67b79ad805ec11f1fb507962b2b6/plugins/Amplifier/Amplifier.cpp#L123-L125